### PR TITLE
Move scheduler's NodeInfo and Listers types to framework pkg

### PR DESCRIPTION
**Pull request by ahg-g**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
Moves scheduler's NodeInfo and SharedLister types to the scheduler's framework pkg. Those types are part of the framework API and so it makes sense to have them in the framework pkg.

**Which issue(s) this PR fixes**:
Part of #89528

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/scheduler/framework/v1alpha1/listers.go [hunk 1]
```diff
@@ -38,11 +37,11 @@ type PodLister interface {
 // NodeInfoLister interface represents anything that can list/get NodeInfo objects from node name.
 type NodeInfoLister interface {
```
Comment (alculquicondor) [inline pkg/scheduler/framework/v1alpha1/listers.go:38]: These types should move to `framework/v1alpha1/interface.go`.
But the actual implementation could go to a separate subpackage (perhaps `framework/v1alpha1/listers`).
The rationale is that plugins and the cluster-autoscaler don't need the implementations.

But followup is fine.

Comment (ahg-g) [reply pkg/scheduler/framework/v1alpha1/listers.go:38]: I don't like placing everything in one giant interface.go file. I agree we need to split the implementation and the interface into separate pkgs, and once in separate packages we should rename ```interface.go``` -> ```framework.go```, ```listers.go``` and ```types.go``` should stay (the WaitingPod, CycleState should move to ```types.go``` and remove their files)

File: pkg/scheduler/core/generic_scheduler_test.go [hunk 2]
```diff
@@ -54,12 +54,11 @@ import (
 	"k8s.io/kubernetes/pkg/scheduler/framework/plugins/volumerestrictions"
 	"k8s.io/kubernetes/pkg/scheduler/framework/plugins/volumezone"
 	framework "k8s.io/kubernetes/pkg/scheduler/framework/v1alpha1"
+	fakelisters "k8s.io/kubernetes/pkg/scheduler/framework/v1alpha1/fake"
```
[the remark this answers is not in the document]
Comment (ahg-g) [reply pkg/scheduler/core/generic_scheduler_test.go]: done.

File: pkg/scheduler/framework/v1alpha1/listers.go (hunk 1 above)
Comment (alculquicondor) [reply pkg/scheduler/framework/v1alpha1/listers.go:38]: sgtm

**Comment by ahg-g** [comment]
@msau42 can you please approve the test related changes, this is part of the refactoring of the scheduler's pkgs, hence no behavioral change.

**Comment by msau42** [comment]
/approve
test changes

**Pull request merged**